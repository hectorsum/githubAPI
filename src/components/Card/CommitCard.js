import React from 'react'
import { Link } from 'react-router-dom';
import styled from 'styled-components'
import { Button } from '../Button/Button';
export const CommitCard = ({sha,commit:{message}}) => {
  return (
    <div className="col-lg-4 col-md-6">
      <Card className="card m-2">
        <CardContent className="card-body">
          <CardTitle>{sha}</CardTitle>
          <CardBody>{message}</CardBody>
          <Underline/>
          <Link to={`./detail/${sha}`}>
            <Button>See more.. ➡️</Button>
          </Link>
        </CardContent>
      </Card>
    </div>
  )
}

const Underline = styled.hr `
  color:white;
`;

const CardBody = styled.p `
  font-size:15px;
  color: ${props => props.theme.paragraphColor};
  font-family: 'Work Sans', sans-serif;
`;

const CardTitle = styled.h3 `
  font-size:16px;
  font-family: 'Work Sans', sans-serif;
  color: ${props => props.theme.titleColor}
`;

const CardContent = styled.div `

`;

const Card = styled.div `
  background-color: ${props => props.theme.cardColor};
`;