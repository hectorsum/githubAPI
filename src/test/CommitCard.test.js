import React from "react";
import "@testing-library/jest-dom";
import { shallow,configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { CommitCard } from "../components/Card/CommitCard";
import { createSerializer } from "enzyme-to-json";

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Testing <CommitCard/>", () => {
  const key = "key";
  const commit = {message: '1'};
  const wrapper = shallow(<CommitCard key={key} commit={commit} />);
  test("Should show the <CommitCard/> component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});