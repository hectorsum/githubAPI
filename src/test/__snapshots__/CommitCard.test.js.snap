// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CommitCard/> Should show the <CommitCard/> component correctly 1`] = `
<div
  className="col-lg-4 col-md-6"
>
  <styled.div
    className="card m-2"
  >
    <styled.div
      className="card-body"
    >
      <styled.h3 />
      <styled.p>
        1
      </styled.p>
      <styled.hr />
      <Link
        to="./detail/undefined"
      >
        <styled.button>
          See more.. ➡️
        </styled.button>
      </Link>
    </styled.div>
  </styled.div>
</div>
`;
